custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -342,9 +344,34 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
     vm.onSelectYear = function (year) {
         var date = new Date();
         var latest = date;
+        vm.years = vm.yearsCopy;
+        vm.months = vm.monthsCopy;
+
         if (vm.isIncentiveReportSelected()) {
-            var offset = date.getDate() < 15 ? 2 : 1;
-            latest.setMonth(date.getMonth() - offset);
+            vm.years = _.filter(vm.yearsCopy, function (y) {
+                return y.id >= 2018;
+            });
+
+            if (vm.selectedYear === latest.getFullYear()) {
+                var offset = date.getDate() < 15 ? 1 : 0;
+
+                vm.months = _.filter(vm.monthsCopy, function (month) {
+                    return month.id <= latest.getMonth() - offset;
+                });
+
+                if (vm.selectedMonth > vm.months.slice(-1)[0].id) {
+                    vm.selectedMonth = vm.months.slice(-1)[0].id;
+                }
+            } else {
+                if (vm.selectedYear === 2018) {
+                    vm.months = vm.months.slice(-3);
+
+                    if (vm.selectedMonth < 10) {
+                        vm.selectedMonth = 12;
```

Comment:
Should not it be set to october. as the ticket says that 'We want the filter options to be available from October 2018'

Reply:
It currently looks that it sets to December.